### PR TITLE
New unit test for AD divergence(gradient(vector))

### PR DESCRIPTION
**Pull request by roystgnr**
This really belongs in https://github.com/roystgnr/MetaPhysicL, but @clarkpede already did all the work to put it here, and even if I get upstream up to snuff it's nice to have belt-and-suspenders testing.

### CODE REVIEW

**Comment by nicholasmalaya** [comment]
Aside from that minor note, this looks great and is ready for merge.

**Comment by roystgnr** [comment]
I missed something.  What minor note?

**Comment by nicholasmalaya** [comment]
Sorry, github has changed the interface to add 'pending' comments before sending a series of comment spam. Good idea but I was aware. Comments should be present now.

File: tests/ad_divgradsimple.cpp [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  typedef ThirdDerivType ADScalar;
+
+  //
+  NumberVector<NDIM, ADScalar> U;
+
+  ADScalar x = ADScalar(x1,NumberVectorUnitVector<NDIM, 0, Scalar>::value());
+  ADScalar y = ADScalar(y1,NumberVectorUnitVector<NDIM, 1, Scalar>::value());
+  ADScalar z = ADScalar(z1,NumberVectorUnitVector<NDIM, 2, Scalar>::value());
+
+  // Assign a simple manufactured solution to U
```
Comment (roystgnr) [inline tests/ad_divgradsimple.cpp]: No, but it's not; this isn't even a PDE solution.  This really is a proper unit test of ```div(grad(U)) == analytic_div_grad_U```, not just a bloated regression test.

**Comment by clarkpede** [comment]
Everything looks good.  All the unit tests are passing on my system.  I applied the navierstokes_incompressible_homogeneous manufactured solution to my DNS code, and the verification tests match what I was previously getting using the analytical (not automatic-differentiation) form.

One minor proposed change.  When I created the unit test, I widened the tolerance so that it would be obvious something was wrong.  You can see this on lines 60-61 of the unit test [https://github.com/manufactured-solutions/MASA/blob/master/tests/ad_divgradsimple.cpp](url):
`  // CCP: I hacked this threshold to be very lenient of errors
  const Scalar thresh = 5000 * std::numeric_limits<Scalar>::epsilon();`
We may want to bring this tolerance back down, but I'm not sure what would be a good tolerance.

**Comment by roystgnr** [comment]
https://github.com/manufactured-solutions/MASA/pull/40 brings us down to 50*epsilon, which ought to be tight enough to catch subtle errors, but I don't think it's important.  That indexing error was giving us O(1) mistakes, not O(epsilon).

**Comment by nicholasmalaya** [comment]
50x \epsilon is well within the spirit of MASA. That 5000 was making me *very* nervous. I'm fine with 50.

**Pull request merged**